tests/unit/backend/data/test_GroceryService.py

```diff
@@ -1217,39 +1214,48 @@ def test_fetch_grocery_list_diffcal_mask_cached(self):
     def test_fetch_grocery_list_diffcal_mask_loads_table(self):
         # Test of workspace type "diffcal_mask" as `Input` argument in the `GroceryList`:
         #   * corresponding table workspace is also loaded from the hdf5-format file.
-        path = Resource.getPath("outputs")
         self.instance._fetchInstrumentDonor = mock.Mock(return_value=self.sampleWS)
-        with tempfile.TemporaryDirectory(dir=path, suffix="/") as tmpPath:
-            self.instance.dataService._constructCalibrationDataPath.return_value = tmpPath
+        oldself = self.instance.dataService._constructStateRoot(self.runNumber1)
+        with RedirectStateRoot(self.instance.dataService) as tmpRoot:
             groceryList = GroceryListItem.builder().native().diffcal_mask(self.runNumber1).buildList()
             diffCalMaskName = wng.diffCalMask().runNumber(self.runNumber1).build()
 
             # DiffCal filename is constructed from the table name
             diffCalTableName = wng.diffCalTable().runNumber(self.runNumber1).build()
-            diffCalTableFilename = diffCalTableName + ".h5"
-            shutil.copy2(self.sampleDiffCalFilePath, Path(tmpPath) / diffCalTableFilename)
-            assert (Path(tmpPath) / diffCalTableFilename).exists()
+            diffCalTableFilename = self.instance._createDiffcalTableFilename(
+                runNumber=groceryList[0].runNumber,
+                useLiteMode=groceryList[0].useLiteMode,
+                version=groceryList[0].version,
+            )
+            tmpRoot.addFileAs(self.sampleDiffCalFilePath, diffCalTableFilename)
+            assert Path(diffCalTableFilename).exists()
 
             assert not mtd.doesExist(diffCalMaskName)
             assert not mtd.doesExist(diffCalTableName)
             items = self.instance.fetchGroceryList(groceryList)
             assert items[0] == diffCalMaskName
             assert mtd.doesExist(diffCalMaskName)
             assert mtd.doesExist(diffCalTableName)
+        # NOTE make sure the state root method is returned to its original pristine form
```

Comment:
We now have both a `state_root_override` context manager, and a `state_root_override_fixture` -- please use whichever is appropriate here.  For a test, the fixture is more reliable in terms of what happens when the test fails.

Reply:
As discussed, my context manager does something a bit different, but because they're related I moved mine to the same helper file and renamed it in a corresponding way.